src/lib/evil/evil_windows.h

```diff
@@ -8,11 +8,9 @@
 #ifndef WIN32_LEAN_AND_MEAN
 # define WIN32_LEAN_AND_MEAN
 #endif
-
-#include <evil_api.h>
-
 #include <windows.h>
-#include <winsock2.h>
+#undef WIN32_LEAN_AND_MEAN
 
+#include <evil_api.h>
```

Comment:
Why putting `evil_api.h` on the place of `winsock2.h`? `evil_api.h` doesn't seam to include `winsock2.h` nor does what `winsock2.h` should do.

Reply:
`evil_api.h` was moved below windows.h and I am not sure about `winsock2` but it seems not always needed. Just to remember, `evil_windows.h` is included in `evil_private.h`.

Reply:
but looking at the diff, it shows that `winsock2.h` was removed.

Reply:
Yes it is removed, do you think it should be always included with `windows.h`?

Reply:
I have no idea about it, but it was included before, so we *expect* it to be necessary.
As this PR just wants to protect `windows.h` of adding unwanted headers, maybe it shouldn't remove this, unless it is causing an error.

Reply:
Agreed. Redundancy is not a problem.

Reply:
My bad. I tested again and in fact `<winsock2.h>` really needs to be removed. I will rebase to keep on the right track.